src/psyclone/tests/domain/lfric/builtins/X_innerproduct_X_test.py

```diff
@@ -0,0 +1,122 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2017-2023, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: A. R. Porter, STFC Daresbury Lab
+# Modified: I. Kavcic, Met Office
+# Modified: R. W. Ford and N. Nobre, STFC Daresbury Lab
+# Modified: by J. Henrichs, Bureau of Meteorology
+
+''' Module containing pytest tests of the LFRicXInnerproductXKern built-ins
+    (inner product of a real-valued field with itself).'''
+
+import os
+from psyclone.domain.lfric.kernel import LFRicKernelMetadata
+from psyclone.domain.lfric.lfric_builtins import LFRicXInnerproductXKern
+from psyclone.parse.algorithm import parse
+from psyclone.psyGen import PSyFactory
+from psyclone.psyir.nodes import Assignment, Loop
+from psyclone.tests.lfric_build import LFRicBuild
+
+# Constants
+BASE_PATH = os.path.join(
+    os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(
+        os.path.abspath(__file__))))),
+    "test_files", "dynamo0p3")
+
+# The PSyclone API under test
+API = "dynamo0.3"
+
+
+# pylint: disable=invalid-name
```

Comment:
I see you had to disable "invalid-name" in all this test files, but this still doesn't remove the pylint issue:
` Module name "X_innerproduct_X_test" doesn't conform to snake_case naming style`
Coming from the filename itself.

I would prefer to conform to python guidelines here and rename the filenames and the testnames in this folder to use lowercase instead of disabling pylint. (To be clear, not the classes or Fortran filenames, this are already properly capitalised)

Reply:
And not the fault of this PR but pycodestyle finds some issues, would you mind fixing them as they are easy fixes:
```
src/psyclone/tests/psyir/transformations/transformations_test.py:113:11: E275 missing whitespace after keyword
src/psyclone/tests/psyir/transformations/transformations_test.py:116:1: W293 blank line contains whitespace
src/psyclone/psyir/nodes/intrinsic_call.py:969:1: E305 expected 2 blank lines after class or function definition, found 1
```